syntax = "proto3";

import "google/protobuf/empty.proto";

package erp;

service OrderService {
  rpc ProcessOrder (OrderRequest) returns (OrderResponse);
  rpc UpdateOrderStatus (OrderStatusUpdateRequest) returns (google.protobuf.Empty);
}

message OrderRequest {
  string order_id = 1;
  string product_id = 2;
  int32 quantity = 3;
}

message OrderResponse {
  string shipping_date = 1;
  string order_status = 2;
}

message OrderStatusUpdateRequest {
  string order_id = 1;
  int32 status = 2;
}
